tests/test_interface.rb

```diff
@@ -165,6 +170,37 @@ def interface_supports_property?(intf, message)
     flunk(message)
   end
 
+  # Helper to find first valid speed that isn't "auto"
+  def valid_speed_set(interface)
+    valid_speed = nil
+    speeds = capable_speed_values(interface)
+    speeds.each do |value|
+      next if value == 'auto'
+      begin
+        interface.speed = value
+        assert_equal(value.to_i, interface.speed)
+      rescue Cisco::CliError => e
+        next if speed_change_disallowed?(e.message)
+        raise
```

Comment:
Keep it. I don't feel strongly about it. 
I think the only difference you'll see is that you'll get a traceback with `raise` versus a simple failure message with `flunk`.